### PR TITLE
Edit three guides to support Cloud users

### PR DESCRIPTION
**Pull request by ptgott**
See #10633

Upgrading

- Misc. grammar/style/clarity tweaks
- Add details re: checking the Cloud Proxy/Auth versions for Cloud
  users
- Add a scoped Tabs component for the upgrade sequence
- Remove the "upgrading to Teleport 4.0+" section since we no longer
  support this version

Backup and restore

- Add scoped Tabs components where instructions vary between editions
- Misc clarity/grammar/style improvements

Authentication

Use Tabs to ensure that readers who have selected one scope don't see
content that is relevant only for other scopes.

### CODE REVIEW

File: docs/pages/setup/operations/backup-restore.mdx [hunk 1]
```diff
@@ -22,35 +30,200 @@ For example, a customer running Teleport on AWS with DynamoDB have these key ite
...
+    
+</TabItem>
+<TabItem scope={["cloud"]} label="Teleport Cloud">
+<Notice type="tip">
+
+In Teleport Cloud, backend data is managed for you automatically. Follow the
+instructions below if you wish to migrate your Teleport Cloud backend to a
+self-hosted one.
+
+</Notice>
```
Comment (ulysseskan) [inline docs/pages/setup/operations/backup-restore.mdx:226]: The `tctl get all` command listed below this notice doesn't work with Teleport Cloud.  You get `ERROR: this request can be only executed by an admin`.

File: docs/pages/setup/operations/backup-restore.mdx [hunk 2]
```diff
@@ -90,13 +264,28 @@ $ tctl --config fresh.yaml get all
...
+## Limitations
+
+The `--bootstrap` flag doesn't re-trigger Trusted Cluster handshakes, so Trusted
+Cluster resources need to be recreated manually.
+
+All the same limitations around modifying the config file of an existing cluster
+also apply to a new cluster being bootstrapped from the state of an old cluster.
 
-**Limitations**
+Of particular note:
```
Comment (ulysseskan) [inline docs/pages/setup/operations/backup-restore.mdx:279]: Can probably leave out the word particular here.

File: docs/pages/setup/operations/upgrading.mdx [hunk 3]
```diff
@@ -5,87 +5,93 @@ description: How to upgrade Teleport components
...
+In Teleport Cloud, we manage the Auth and Proxy Services for you. You can
+determine the current version of these services by running the following
+command, where `mytenant` is the name of your Teleport Cloud tenant:
 
-- **Only patch** versions are always compatible, for example, any 4.0.1 component will work with any 4.0.3 component.
-- Minor versions are always compatible with the **previous** minor release. This means you must not attempt to upgrade from 4.1.x straight to 4.3.x. You must upgrade to 4.2.x first.
-- Teleport clients [`tsh`](../reference/cli.mdx#tsh) for users and [`tctl`](../reference/cli.mdx#tctl) for admins
-  may not be compatible with different versions of the `teleport` service.
+```code
+$ curl -s https://mytenant.teleport.sh/webapi/ping | jq '.server_version'
```
Comment (ulysseskan) [inline docs/pages/setup/operations/upgrading.mdx:29]: This is handy - thanks for including.

File: docs/pages/setup/operations/backup-restore.mdx [hunk 4]
```diff
@@ -217,75 +215,13 @@ Of particular note:
...
 <TabItem scope={["cloud"]} label="Teleport Cloud">
-<Notice type="tip">
 
-In Teleport Cloud, backend data is managed for you automatically. Follow the
-instructions below if you wish to migrate your Teleport Cloud backend to a
-self-hosted one.
+In Teleport Cloud, backend data is managed for you automatically. If you would
+like to migrate configuration resources to a self-hosted Teleport cluster,
+follow our recommended backup practice of storing configuration resources in a
+git repository and creating them regularly. This will enable you to keep your
```
Comment (ulysseskan) [inline docs/pages/setup/operations/backup-restore.mdx:222]: I'm not sure if this should say "creating them regularly" or "updating them regularly".

Comment (ptgott) [reply docs/pages/setup/operations/backup-restore.mdx]: Added another suggestion that's less ambiguous

**Review by ulysseskan — APPROVED** [review]

File: docs/pages/setup/reference/authentication.mdx [hunk 5]
```diff
@@ -27,44 +27,143 @@ connector. There are several possible values (types) of 2FA:
...
+metadata:
+  name: cluster-auth-preference
+spec:
+  type: saml
+version: v2
+```
+
+### OIDC
+
+Teleport implements OpenID Connect (OIDC) authentication, which is similar to SAML in principle.
```
Comment (zmb3) [inline docs/pages/setup/reference/authentication.mdx:150]: I don't think we need to mention "similar to SAML here."

Those who don't know the technical details probably don't care what it's similar to, and those who do know may or may not object to this.

I don't think we lose anything by just omitting this part. If we really do want to compare to some other option, I would move OIDC before SAML and after Github, and mention that OIDC is similar to OAuth (it's basically an extension to OAuth 2.0).

**Review by zmb3 — APPROVED** [review]

File: docs/pages/setup/operations/upgrading.mdx [hunk 6]
```diff
@@ -5,87 +5,93 @@ description: How to upgrade Teleport components
...
 - **Patch and minor** versions are always compatible, for example, any 5.0.1 component will work with any 5.0.3 component and 6.1.0 component will work with any 6.7.0 component.
 - Major versions are always compatible with the **previous** major release. This means you must not attempt to upgrade from 5.x.x straight to 7.x.x. You must upgrade to 6.x.x first.
-- The above applies to both clients and servers. For example, a 6.x.x proxy is
-  compatible with 5.x.x nodes and 5.x.x `tsh`. But we don't guarantee that a
-  7.x.x `tsh` will work with a 5.x.x proxy.
+- The above applies to both clients and servers. For example, a 6.x.x Proxy Service is
+  compatible with 5.x.x Nodes and 5.x.x `tsh`. But we don't guarantee that a
+  7.x.x `tsh` will work with a 5.x.x Proxy Service.
+
+### Before 5.0.0
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/setup/operations/upgrading.mdx:49]: I'll remove it in the next push

File: docs/pages/setup/reference/authentication.mdx [hunk 7]
```diff
@@ -27,44 +27,143 @@ connector. There are several possible values (types) of 2FA:
   the value to `on`.
 - `off` turns off second-factor authentication.
 
-You can modify these settings in the static config `teleport.yaml` or using dynamic configuration resources:
+<Admonition type="note">
+  If you are using Teleport with a Single Sign-On solution, users can also register 2FA devices, but Teleport will not prompt them for 2FA during login. 2FA for SSO users should be handled by the SSO provider.
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/setup/reference/authentication.mdx:31]: I've swapped out `2FA` with `MFA`. In the future, I want to include guidelines for decisions like this in our style guide.

**Pull request merged**